### PR TITLE
Add `LTE` instruction.

### PR DESCRIPTION
**Pull request by thealmarty**
- closes https://github.com/lita-xyz/llvm-valida/issues/63
- for the STARK, I just added an extra output constraint: the output should be 1 if `is_lte` & `input_1 == input_2`
- now all tests pass when one runs `bash llvm-test-suite/run-valida-tests.sh`

### CODE REVIEW

**Review by morganthomas — CHANGES_REQUESTED** [review]
This should build in CI once you merge `main` into it. This would apply the correct Plonky3 version pin in CI. I was able to build it locally. All the tests passed except for `prove_fibonacci`. This one failed for me like this:

```
failures:

---- prove_fibonacci stdout ----
thread 'prove_fibonacci' panicked at /home/morgan/code/lita/issue/lte/valida/machine/src/quotient.rs:41:35:
attempt to subtract with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    prove_fibonacci

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.51s
```

Not sure what happened here, but am willing to help to track it down.

**Comment by thealmarty** [comment]
Thanks Morgan. Now that I rebased I can see the same error you mentioned. Looking into it.

File: alu_u32/src/lt/mod.rs [hunk 1]
```diff
@@ -83,37 +90,51 @@ impl Lt32Chip {
...
-                cols.output = F::from_canonical_u8(dst[3]);
-                cols.multiplicity = F::one();
+            Operation::Lt32(a, b, c) => {
+                cols.is_lt = F::one();
+                self.set_cols(cols, a, b, c);
+            }
+            Operation::Lte32(a, b, c) => {
+                cols.is_lte = F::one();
+                self.set_cols(cols, a, b, c);
             }
```
[the remark this answers is not in the document]
Comment (thealmarty) [reply alu_u32/src/lt/mod.rs:100]: `a`, `b`, `c` have local scopes so it seems repetition is necessary unfortunately.

[the remark this answers is not in the document]
Comment (morganthomas) [reply alu_u32/src/lt/mod.rs:100]: Ah, good point.

**Comment by thealmarty** [comment]
I changed the constraint with the latest commit. It doesn't fail with `cargo test` and the compiler tests. Please review.

**Review by morganthomas — APPROVED** [review]

**Pull request merged**